src/arch/runtime/context_switching.cc

```diff
@@ -292,7 +289,7 @@ artificial_stack_t::artificial_stack_t(void (*initial_fun)(void), size_t _stack_
 #elif defined(__s390x__)
     sp -= 16; // r6-r13 and f8-f15.
 #elif defined (__powerpc64__)
-    sp += 24;
+    sp -= 20; // r14-r31, toc, cr.
```

Comment:
Strictly speaking you also need to save f14-f31 and v20-v31 to meet the ABI requirements. The compiler could choose to save something in one of these registers and a coroutine switch might end up clobbering it. It's the sort of thing that is very hard to debug. At the very least there should be a comment here explaining the danger.